rfcs/0136-scope-expression-registration.md

```diff
@@ -0,0 +1,129 @@
+# RFC 136 - Scope Expression Registration
+* Comments: [#136](https://api.github.com/repos/taskcluster/taskcluster-rfcs/issues/136)
+* Proposed by: @imbstack
+
+# Summary
+
+There should be a way that someone administering Taskcluster (e.g. adding scopes to a role) to know
+what a set of scopes allow something holding them to "do" within Taskcluster. This should be published
+in a format that allows others to build tools/tests on top of this.
+
+## Motivation
+
+It is currently possible to figure some subset of this out with some background knowledge of the
+platform and docs/code reading, but is in general too difficult. In addition, there are
+workers/services that do not ever publish the scopes that they use
+for access control, making this task yet more difficult.
+
+This issue could be partially solved just by building easier ways for services/workers to publish docs
+but it will provide no guarantee that any docs published this way are valid or current. We
+should build a system that allows for these to be positively asserted.
+
+Protecting the registration system by scopes and making it the only way to do authorization allows
+a cluster admin to pick and choose which services will interact with their cluster and also allow
+them to know at any given time which services exist in their cluster.
+
+In addition, this will have the benefits of moving us to doing authorization in the auth service and
+ensure that this move is performant as this will drastically reduce the amount of bits we need to
+shovel back and forth over the wire than would be true otherwise.
+
+# Details
+
+We will define the following:
+
+* **scope expression:** A way of building complex assertions from basic scopes. This is currently defined
+  in taskcluster-lib-scopes. It will need to be standardized more generally for this to be an external api.
+* **scope expression template:** A method of building a scope expression given certain parameters. Until
```

Comment:
To be clear, this is in the simplest terms,`resource:<x>` vs `resource:manatee`?  Which level do things like ifs and fors fit into?

Reply:
`if` and `for` are at the template level. The expressions themselves are just maps with strings in them. No substitution, etc.

I'll update the definitions to express that.